src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMFeatureSelector.h

```diff
@@ -0,0 +1,201 @@
+      const std::map<String, std::vector<Feature>>& feature_name_map,
+      std::vector<String>& result
+    ) = 0;
+
+    /**
+      Splits the features into time segments. `optimize()` method is run on each of these segments.
+
+      @param[in] features Input features
+      @param[out] selected_filtered Output features
+    */
+    void select_MRMFeature(const FeatureMap& features, FeatureMap& selected_filtered);
+
+    void setNNThreshold(const Int nn_threshold);
+    Int getNNThreshold() const;
+
+    void setLocalityWeight(const bool locality_weight);
+    bool getLocalityWeight() const;
+
+    void setSelectTransitionGroup(const bool select_transition_group);
+    bool getSelectTransitionGroup() const;
+
+    void setSegmentWindowLength(const Int segment_window_length);
+    Int getSegmentWindowLength() const;
+
+    void setSegmentStepLength(const Int segment_step_length);
+    Int getSegmentStepLength() const;
+
+    void setVariableType(const String& variable_type);
+    String getVariableType() const;
+
+    void setOptimalThreshold(const double optimal_threshold);
+    double getOptimalThreshold() const;
+
+    void setScoreWeights(const std::map<String, String>& score_weights);
+    std::map<String, String> getScoreWeights() const;
+
+protected:
+    /// Add variable to the LP problem instantiated in `optimize()`
+    Int addVariable(LPWrapper& problem, const String& name, const bool bounded = true, const double obj = 1.0) const;
+
+    /// Scoring method used by the optimizer. Based off score weights parameter. The returned value is used for the LP problem's variables and contraints.
+    double compute_score(const Feature& feature) const;
+
+    /// Add constraint to the LP problem instantiated in `optimize()`
+    void addConstraint(
+      LPWrapper& problem,
+      std::vector<Int> indices,
+      std::vector<double> values,
+      const String& name,
+      const double lb,
+      const double ub,
+      const LPWrapper::Type param
+    ) const;
+
+private:
+    Int    nn_threshold_            = 4;
+    bool   locality_weight_         = false;
+    bool   select_transition_group_ = true;
+    Int    segment_window_length_   = 8;
+    Int    segment_step_length_     = 4;
+    String variable_type_           = "continuous";
+    double optimal_threshold_       = 0.5;
+    std::map<String, String> score_weights_;
+
+    // TODO: rename method
+    void constructToList(
+      const FeatureMap& features,
+      std::vector<std::pair<double, String>>& time_to_name,
+      std::map<String, std::vector<Feature>>& feature_name_map
+    ) const;
+
+    /// Transform the given score thorugh the chosen lambda function
+    double weight_func(const double score, const String& lambda_score) const;
+
+    /// Helper method to remove spaces from a string
+    String remove_spaces(String str) const;
+  };
+
+  class OPENMS_DLLAPI MRMFeatureSelectorQMIP : public MRMFeatureSelector
```

Comment:
Class used to select MRMFeatures based on relative retention time using a quadratic mixed integer programming (QMIP) formulation.  The method is described in [TODO: update when published]